src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -353,21 +353,21 @@ private static MetadataReference GetOrCreateMetadataReference(ref MetadataRefere
         {
             return Diagnostic(
                 code,
-                NormalizeDiagnosticString(squiggledText.Value),
+                NormalizeNewLines(squiggledText),
                 arguments,
                 startLocation,
                 syntaxNodePredicate,
                 argumentOrderDoesNotMatter);
         }
 
-        protected static string NormalizeDiagnosticString(string inputString)
+        public static string NormalizeNewLines(XCData data)
         {
-            if (!inputString.Contains("\r\n") && inputString.Contains("\n"))
+            if (ExecutionConditionUtil.IsWindows)
```

Comment:
Isn't this going to break any text that comes in with `\r\n`? We'll end up with `\r\r\n`.

Reply:
The `XCData` type is the input here and it can only have `\n` in the `Value` property. Hence this is safe